lib/posix_channel.ml

```diff
@@ -190,8 +190,14 @@ let receive protocols =
   | V4V_proxy(_, _) -> assert false (* weight is 0 above *)
   | TCP_proxy(ip, port) ->
     let s = Unix.socket Unix.PF_INET Unix.SOCK_STREAM 0 in
-    Unix.connect s (Unix.ADDR_INET(Unix.inet_addr_of_string ip, port));
-    s
+    begin
+      try
+        Unix.connect s (Unix.ADDR_INET(Unix.inet_addr_of_string ip, port));
```

Comment:
Use `Xapi_stdext_pervasives.Pervasiveext.finally`, it' is used also in other parts of the code: `finally (fun () -> Unix.connect s (Unix.ADDR_INET(Unix.inet_addr_of_string ip, port)); s) (fun () -> Unix.close s)` it will take care of re-rising